compiler/hash-ir/src/ir.rs

```diff
@@ -16,6 +15,10 @@ use hash_storage::{
         SequenceStore, SequenceStoreKey,
     },
 };
+use hash_tir::intrinsics::definitions::{
+    BinOp as TirBinOp, CondBinOp as TirCondBinOp,
+    ShortCircuitingBoolOp as TirShortCircuitingBoolOp, UnOp as TirUnOp,
+};
```

Comment:
Haha, verbose naming 💯  

I feel like we should unify operators in `hash-source`, especially since we will kind of need to do that if we want to share using the `Const` representation across the compiler. Perhaps, even define it in `hash-token`?

It would eliminate:
- `ir::BinOp` `ir::UnOp`
- `ast::BinOp` `ast::UnOp`
- `tir::BinOp`  `tir::UnOp`

But maybe the separation makes sense?

Reply:
I think it's a bit different because the operators defined in the AST do not necessarily correlate with the "intrinsic" operations that can be defined on TIR terms. For example, with the new constant stuff a lot of the intrinsics will greatly simplify/become redundant. I think I will first adapt to `Const` and then we can consider this separation more.

Reply:
When we do integrate the new constant stuff, we will still have to define some kind of common operator kinds, I guess for now the IR can just rely on them and the TIR can define conversions for them.

Reply:
Okay. I agree that there should be some unification here..